flags.go

```diff
@@ -142,23 +185,113 @@ func (g *goforit) Enabled(ctx context.Context, name string) (enabled bool) {
 
 	g.flagsMtx.RLock()
 	defer g.flagsMtx.RUnlock()
+	enabled = false
 	if g.flags == nil {
-		enabled = false
+		log.Printf("[goforit] empty flags map")
 		return
 	}
 	lastRefreshTime = g.lastFlagRefreshTime
 	flag := g.flags[name]
 
-	// equality should be strict
-	// because Float64() can return 0
-	if f := g.rand(); f < flag.Rate {
-		enabled = true
+	if !flag.Active {
 		return
 	}
+
+	var mergedProperties = map[string]string{}
+
+	for k, v := range g.defaultTags {
+		mergedProperties[k] = v
+	}
+
+	for k, v := range properties {
+		mergedProperties[k] = v
+	}
+
+	for _, r := range flag.Rules {
+		res, err := r.Rule.Handle(flag.Name, mergedProperties)
+		if err != nil {
+			log.Printf("[goforit] error evaluating rule:\n", err)
+			return
+		}
+		var matchBehavior RuleAction
+		if res {
+			matchBehavior = r.OnMatch
+		} else {
+			matchBehavior = r.OnMiss
+		}
+		switch matchBehavior {
+		case "on":
+			enabled = true
+			return
+		case "off":
+			enabled = false
+			return
+		case "continue":
+			continue
+		default:
+			log.Printf("[goforit] unknown match behavior: " + string(matchBehavior))
```

Comment:
Things like this should probably be some kind of metric too? But let's punt on that for now, maybe it's a good case for Sentry integration.

Reply:
Also, we should use the logger inside `g *goforit` instead of just a basic log.

Reply:
changed to g.logger, but yeah would love to tackle Sentry before we ship!